metrics/mean_iou/mean_iou.py

```diff
@@ -0,0 +1,237 @@
+    total_area_pred_label = np.zeros((num_classes,), dtype=np.float64)
+    total_area_label = np.zeros((num_classes,), dtype=np.float64)
+    for result, gt_seg_map in zip(results, gt_seg_maps):
+        area_intersect, area_union, area_pred_label, area_label = intersect_and_union(
+            result, gt_seg_map, num_classes, ignore_index, label_map, reduce_zero_label
+        )
+        total_area_intersect += area_intersect
+        total_area_union += area_union
+        total_area_pred_label += area_pred_label
+        total_area_label += area_label
+    return total_area_intersect, total_area_union, total_area_pred_label, total_area_label
+
+
+def mean_iou(
+    results, gt_seg_maps, num_classes, ignore_index, nan_to_num=None, label_map=dict(), reduce_zero_label=False
+):
+    """Calculate Mean Intersection and Union (mIoU)
+
+    Args:
+        results (list[ndarray]):
+            List of prediction segmentation maps.
+        gt_seg_maps (list[ndarray]):
+            List of ground truth segmentation maps.
+        num_classes (int):
+            Number of categories.
+        ignore_index (int):
+            Index that will be ignored in evaluation.
+        nan_to_num (int, optional):
+            If specified, NaN values will be replaced by the numbers defined by the user. Default: None.
+        label_map (dict):
+            Mapping old labels to new labels. Default: dict().
+        reduce_zero_label (bool):
+            Whether to ignore the zero label. Default: False.
+
+     Returns:
+        dict[str, float | ndarray]:
+            <aAcc> float: Overall accuracy on all images.
+            <Acc> ndarray: Per category accuracy, shape (num_classes, ).
+            <IoU> ndarray: Per category IoU, shape (num_classes, ).
+    """
+    iou_result = eval_metrics(
+        results=results,
+        gt_seg_maps=gt_seg_maps,
+        num_classes=num_classes,
+        ignore_index=ignore_index,
+        metrics=["mIoU"],
+        nan_to_num=nan_to_num,
+        label_map=label_map,
+        reduce_zero_label=reduce_zero_label,
+    )
+    return iou_result
+
+
+@datasets.utils.file_utils.add_start_docstrings(_DESCRIPTION, _KWARGS_DESCRIPTION)
+class Mean_IoU(datasets.Metric):
+    def _info(self):
+        return datasets.MetricInfo(
+            description=_DESCRIPTION,
+            citation=_CITATION,
+            inputs_description=_KWARGS_DESCRIPTION,
+            features=datasets.Features(
+                {
+                    "predictions": datasets.Sequence(datasets.Value("int32")),
+                    "gt_seg_maps": datasets.Sequence(datasets.Value("int32")),
+                    "num_classes": datasets.Value("int32"),
+                    "ignore_index:": datasets.Value("int32"),
+                    "nan_to_num": datasets.Value("int32"),
+                    "label_map": dict,
+                    "reduce_zero_label": datasets.Value("bool"),
+                }
+            ),
```

Comment:
It only makes sense to save predictions  and references to the arrow file.
```suggestion
            features=datasets.Features(
                {
                    "predictions": datasets.Sequence(datasets.Sequence(datasets.Sequence(datasets.Value("int64")))),
                    "references": datasets.Sequence(datasets.Sequence(datasets.Sequence(datasets.Value("int64")))),
                }
            ),
```

Reply:
Oh I see.